lib/components/form/batch-settings.tsx

```diff
@@ -100,7 +115,7 @@ class BatchSettings extends Component<{
 }> {
   state = {
     expanded: null,
-    selectedModes: this.props.modeOptions.map((m) => m.mode)
+    selectedModes: getDefaultModes(this.props.modeOptions)
```

Comment:
As mentioned in the previous review, this assignment to `state` needs to be done in the class constructor. As written, it seems that `this.props` is not properly set at the time of execution.

Reply:
Not sure how this one slipped through! Fixed in 820468c62705513a5cab0b9e635fd0ab50b1d54c